### PR TITLE
NOISSUE - Decouple mainflux/mproxy and mainflux/mainflux

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

### CODE REVIEW

**Review by dborovcanin — COMMENTED** [review]
We do not need to implement a logger. It's OK to use the Mainflux logger in examples - it's a separate package anyways. So, we only need a logger interface specified. The same goes for errors - error handling in mProxy is rather simple - can we avoid having a copy of the Mainflux errors package?

**Comment by manuio** [comment]
@dborovcanin I'm not sure to understand. Logger is also used in the main.go. About errors package, we are using `New` and `Wrap` methods. Should we remove non used methods like `Contains`?

**Comment by dborovcanin** [comment]
For logger, I meant simply defining the logger interface and taking implementation from Mainflux. In examples and in `main.go`, it's ok to have MF dependency because that's not something anyone will import.
For errors, I thought about completely removing that package and relying on pure Go errors because error handling is rather simple. Instead of `wrap`, for example, we can do something like (in session.go):
```go
func wrap(err error, dir direction) error {
	if err == io.EOF {
		return err
	}
	switch dir {
	case up:
		return errors.New(errClient.Error() + ":" + err.Error())
	case down:
		return errors.New(errBroker.Error() + ":" + err.Error())
	default:
		return err
	}
}
```

File: pkg/mqtt/mqtt.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func (p Proxy) ListenTLS(tlsCfg *tls.Config) error {
 
 	l, err := tls.Listen("tcp", p.address, tlsCfg)
 	if err != nil {
-		return errors.Wrap(errCreateListener, err)
+		return errors.New(errCreateListener.Error() + ":" + err.Error())
```
Comment (dborovcanin) [inline pkg/mqtt/mqtt.go:93]: No need for this `errCreateListener`, return an error directly.

File: pkg/session/session.go [hunk 2]
```diff
@@ -129,11 +130,14 @@ func (s *Session) notify(pkt packets.ControlPacket) {
 }
 
 func wrap(err error, dir direction) error {
+	if err == io.EOF {
+		return err
+	}
 	switch dir {
 	case up:
-		return errors.Wrap(errClient, err)
+		return errors.New(errClient.Error() + ":" + err.Error())
```
Comment (dborovcanin) [inline pkg/session/session.go:138]: No need for errors, you can use strings constants instead like:
```go
return errors.New(errClient + err.Error())
```
I quickly benchmarked `errors.New(errClient + err.Error())` vs `fmt.Errorf` and for this simple case, it looks like simple `+` is the most efficient.2

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**